rotkehlchen/premium/premium.py

```diff
@@ -42,31 +43,44 @@ class RemoteMetadata(NamedTuple):
     data_size: int
 
 
+DEFAULT_ERROR_MSG = 'Failed to contact rotki server. Check logs for more details'
+DEFAULT_OK_CODES = (HTTPStatus.OK, HTTPStatus.UNAUTHORIZED, HTTPStatus.BAD_REQUEST)
+
+
 def check_response_status_code(
         response: requests.Response,
         status_codes: Sequence[HTTPStatus],
-        user_msg: str = 'Failed to contact rotki server. Check logs for more details',
+        user_msg: str = DEFAULT_ERROR_MSG,
 ) -> None:
     """
     Check the rotki.com response and if status code is not in the expected list
     log the error and raise RemoteError
     """
     if response.status_code not in status_codes:
         log.error(
-            'rotki server responded with an error response '
+            f'rotki server responded with an error response to {response.url} '
             f'{response.status_code=} and {response.text=}',
         )
         raise RemoteError(user_msg)
 
 
-def _process_dict_response(response: requests.Response) -> dict:
+def _process_dict_response(
```

Comment:
Gave same arguments as `check_response_status_code` here to provide flexibility to the caller to call it however they like and not have to repeat the call.